build/lib/i18n.ts

```diff
@@ -308,4 +542,556 @@ export function processNlsFiles(opts:{fileHeader:string;}): ThroughStream {
+						method: 'GET'
+					};
+
+					let request = http.request(options, (res) => {
+							let xlfBuffer: string = '';
+							res.on('data', (data) => xlfBuffer += data);
+							res.on('end', () => {
+								if (res.statusCode === 200) {
+									stream.emit('data', new File({ contents: new Buffer(xlfBuffer) }));
+								} else {
+									log('Error:', `${slug} in ${project} returned no data. Response code: ${res.statusCode}.`);
+								}
+								translationsRetrieved++;
+							});
+					}).on('error', (err) => {
+						log('Error:', `Failed to query resource ${slug} with the following error: ${err}`);
+					});
+					request.end();
+				});
+			});
+		}
+
+		callback();
+	});
+}
+
+export function prepareJsonFiles(): ThroughStream {
+	return through(function(xlf: File) {
+		let stream = this;
+
+		XLF.parse(xlf.contents.toString()).then(
+			function(resolvedFiles) {
+				resolvedFiles.forEach(file => {
+					let messages = file.messages, translatedFile;
+
+					// ISL file path always starts with 'build/'
+					if (file.originalFilePath.startsWith('build/')) {
+						const defaultLanguages = { 'zh-cn': true, 'zh-tw': true, 'ko': true };
+						if (path.basename(file.originalFilePath) === 'Default' && !defaultLanguages[file.language]) {
+							return;
+						}
+
+						translatedFile = createIslFile('..', file.originalFilePath, messages, iso639_2_to_3[file.language]);
+					} else {
+						translatedFile = createI18nFile(iso639_2_to_3[file.language], file.originalFilePath, messages);
+					}
+
+					stream.emit('data', translatedFile);
+				});
+			},
+			function(rejectReason) {
+				log('Error:', rejectReason);
+			}
+		);
+	});
+}
+
+export function createI18nFile(base: string, originalFilePath: string, messages: Map<string>): File {
+	let content = [
+		'/*---------------------------------------------------------------------------------------------',
+		' *  Copyright (c) Microsoft Corporation. All rights reserved.',
+		' *  Licensed under the MIT License. See License.txt in the project root for license information.',
+		' *--------------------------------------------------------------------------------------------*/',
+		'// Do not edit this file. It is machine generated.'
+	].join('\n') + '\n' + JSON.stringify(messages, null, '\t').replace(/\r\n/g, '\n');
+
+	return new File({
+		path: path.join(base, originalFilePath + '.i18n.json'),
+		contents: new Buffer(content, 'utf8')
+	});
+}
+
+
+const languageNames: Map<string> = {
+	'chs': 'Simplified Chinese',
+	'cht': 'Traditional Chinese',
+	'kor': 'Korean'
+};
+
+const languageIds: Map<string> = {
+	'chs': '$0804',
+	'cht': '$0404',
+	'kor': '$0412'
+};
+
+const encodings: Map<string> = {
+	'chs': 'CP936',
+	'cht': 'CP950',
+	'jpn': 'CP932',
+	'kor': 'CP949',
+	'deu': 'CP1252',
+	'fra': 'CP1252',
+	'esn': 'CP1252',
+	'rus': 'CP1251',
+	'ita': 'CP1252'
+};
+
+export function createIslFile(base: string, originalFilePath: string, messages: Map<string>, language: string): File {
```

Comment:
Did you consider that for the ISL we only need to push ch* and kor. For all other languages we have translations. Or is this not possible since languages are defined by a Transifex project. If so we might want to consider not pushing the Default.isl to Transifex